<launch>
    <arg name="fcu_type" default="1" /> <!-- 0: PX4 Autopilot | 1: Ardupilot Copter -->
    <arg name="takeoff_height" default="4"/>
    <arg name="max_acc" default="10" />
    <arg name="enable_takeoff" default="true" />
    <arg name="mav_yaw" default="0"/>
    
    <node pkg="geometric_controller" type="geometric_controller_node" name="geometric_controller" output="screen">
            <param name="fcu_type" value="$(arg fcu_type)"/>
            <param name="to_hgt" value="$(arg takeoff_height)"/>
            <remap from="command/bodyrate_command" to="/mavros/setpoint_raw/attitude"/>
            <param name="ctrl_mode" value="2" />
            <param name="enable_gazebo_state" value="true"/>
            <!-- <param name="velocity_yaw" value="false"/> -->
            <param name="max_acc" value="$(arg max_acc)" />
            <param name="Kp_x" value="8" />
            <param name="Kp_y" value="8" />
            <param name="Kp_z" value="15" />
            <param name="Kv_x" value="4" />
            <param name="Kv_y" value="4" />
            <param name="Kv_z" value="10" />
            <param name="normalizedthrust_constant" value="0.04"/>
            <param name="normalizedthrust_offset" value="0.0"/>
            <param name="init_pos_x" value="0" />
            <param name="init_pos_y" value="0" />
            <param name="init_pos_z" value="2.5" />
            <param name="attctrl_constant" value="0.1"/>
            <param name="posehistory_window" value="500"/>
            <param name="drag_dx" value="0.0"/>
            <param name="drag_dy" value="0.0"/>
            <param name="drag_dz" value="0.0"/>
            <param name="velocity_yaw" value="false"/>
            <param name="takeoff_enable" value="$(arg enable_takeoff)"/> 
            <param name="mav_yaw" value="$(arg mav_yaw)"/>           
    </node>
    <!-- This node converts HKUST proprietry trajectory.msg to MDFJT.msg-->
    <node pkg="interiit21-drdo-iitm" name="traj_msg_converter" type="trajectory_msg_converter.py" output="screen" launch-prefix="bash -c 'sleep 5; $0 $@'">
        <param name="fast_planner_traj_topic" value="planning/ref_traj"/>
        <param name="traj_pub_topic" value="command/trajectory"/>
    </node>
</launch>